index.bs

```diff
@@ -2712,7 +2591,181 @@ partial interface MLGraphBuilder {
+        1. Make a request to the underlying platform to:
+            1. Create an [=implementation-defined=] platform operator |expandImpl| for this method, given |input| and |newShape|.
+            1. Store a reference of |expandImpl| in |output|.{{MLOperand/[[operator]]}}.
+            1. Create an [=implementation-defined=] platform operand |outputImpl| to represent output,given |output| and |expandImpl|.
+            1. Store a reference to |outputImpl| in |output|.{{MLOperand/[[operand]]}}.
+        1. Connect |input| as input to |expandImpl|.
+        1. Connect |output|.{{MLOperand/[[operand]]}} as output to |expandImpl|.
+    1. Return |output|.
+  </div>
+</details>
+
+### gather ### {#api-mlgraphbuilder-gather}
+Gather values of the input tensor along an axis according to the indices.
+<script type=idl>
+dictionary MLGatherOptions {
+  unsigned long axis = 0;
+};
+
+partial interface MLGraphBuilder {
+  MLOperand gather(MLOperand input, MLOperand indices, optional MLGatherOptions options = {});
+};
+</script>
+
+{{MLGatherOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLGatherOptions>
+    : <dfn>axis</dfn>
+    ::
+        An {{unsigned long}} scalar specifying the axis along which the gathered values are obtained. Its value must be in the range [0, N-1] where N is the [=rank=] of the input tensor.
+</dl>
+
+<div>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input N-D tensor from which the values are gathered.
+        - *indices*: an {{MLOperand}}. The indices N-D tensor of the input values to gather. The values must be in the range [0, N] where N is the value of the input shape indexed by *options.axis*.
+        - *options*: an optional {{MLGatherOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output N-D tensor of [=rank=] equal to M + N - 1 where M and N are the [=rank=] of the *input* and *indices* tensor respectively.
+</div>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>gather(|input|, |indices|, |options|)</dfn> method steps are:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |input| and |indices| is {{MLOperand}}.
+    1. Let |shapeInput| be |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} and |rankInput| the [=list/size=] of |shapeInput|.
+    1. Let |shapeIndices| be |indices|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}.
+    1. Let |axis| be |options|.{{MLGatherOptions/axis}}.
+    1. Let |axisSize| be |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[|axis|]
+    1. If |axis| is greater than or equal to |rankInput|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+    1. [=map/For each=] |index| &rarr; |value| of |indices|:
+        1. If |index| is greater than or equal to |axisSize|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
```

Comment:
The `index` value of `indices` may not be available at the graph build time unless the `indices` is a constant operand.

It may also define the run-time behavior if the `index` value of `indices` is greater than or equal to `axisSize`.

Reply:
Can you propose a wording to add here to clarify that `indices` is a constant operand?

Reply:
It's hard to tell whether an MLOperand is a constant or not. The [current algorithm](https://www.w3.org/TR/webnn/#dom-mlgraphbuilder-constant) of constant only creates an implementation-defined platform operand and sets values to it.

Reply:
Opened a separate issue to track: https://github.com/webmachinelearning/webnn/issues/486. Resolving this one.